src/register/RegistrationPage.jsx

```diff
@@ -147,31 +116,98 @@ class RegistrationPage extends React.Component {
     return true;
   }
 
+  getOptions = () => {
+    const { intl } = this.props;
+    return {
+      countryOptions: [{
+        value: '',
+        label: intl.formatMessage(messages['registration.country.label']),
+      }].concat(getCountryList(getLocale()).map(({ code, name }) => ({ value: code, label: name }))),
+      yearOfBirthOptions: [{
+        value: '',
+        label: intl.formatMessage(messages['registration.year.of.birth.label']),
+      }].concat(YEAR_OF_BIRTH_OPTIONS),
+      educationLevelOptions: EDUCATION_LEVELS.map(key => ({
+        value: key,
+        label: intl.formatMessage(messages[`registration.field.education.levels.${key || 'label'}`]),
+      })),
+      genderOptions: GENDER_OPTIONS.map(key => ({
+        value: key,
+        label: intl.formatMessage(messages[`registration.field.gender.options.${key || 'label'}`]),
+      })),
+    };
+  };
+
+  getOptionalFields() {
```

Comment:
Can we move this into a separate component `OptionalFields` instead of a function to make this code cleaner and shorter?

Reply:
how will we get it's value to send in form payload? We will react to use redux for it. wouldn't it make it complicated? Let me know if I am thinking in the wrong direction

Reply:
Now that I think about it, I think it can be achieved by just sending onChange event to the component. Let me refactor the code